### PR TITLE
Add wind power potential and wind profile indicators

### LINKED ISSUE #1458: Indicator for wind power
### Addressing a Problem?

Add indicator allowing the computation of wind power potential. 

### Potential Solution

Here is one  [ref](https://www.researchgate.net/publication/260022888_An_analysis_of_wind_energy_potential_and_economic_evaluation_in_Zahedan_Iran) which fits wind speed to a Weibull distribution, then integrates over all wind speeds to get the potential assuming power is proportionnal to V cubed. 

In practice, there are cutoff speeds (25 m/s) after which power production drops to 0. See https://www.nrel.gov/docs/fy14osti/61714.pdf

### Additional context

Wind speed has often a diurnal component, so not clear how accurate computations based on daily values would be. Probably best to mandate hourly wind speeds here, but recommendations from knowledgeable folks would be useful. 

### Contribution

- [X] I would be willing/able to open a Pull Request to contribute this feature.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1458 
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds two indices and their indicators: wind_profile and wind_power_potential. 

### Does this PR introduce a breaking change?
No

### Other information:
Ideally we`d add an indicator to compute air density, which is an optional input to the wind power potential.

### CODE REVIEW

File: xclim/indices/_conversion.py [hunk 1]
```diff
@@ -2017,3 +2019,157 @@ def mean_radiant_temperature(
...
+        0,  &  v < u_i \\
+        (v - u_i)^3 / (u_r - u_i)^3,  & u_i ≤ v < u_r \\
+        1, & u_r ≤ v < u_o \\
+        0, & v ≥ u_o
+        \end{cases}
+
+    For non-standard air density (:math:`\rho`), the wind speed is scaled using
+    :math:`v_n = v \left( \frac{\rho}{\rho_0} \right)^{1/3}`.
+
+    Note that the temporal resolution of the wind speed time series has a significant influence on the results,
```
Comment (laura-vanvliet) [inline xclim/indices/_conversion.py:2139]: Perhaps this is not necessary, but you could also note that temporal resolution may affect aggregation in time

File: xclim/indices/_conversion.py [hunk 2]
```diff
@@ -2017,3 +2019,157 @@ def mean_radiant_temperature(
...
+    wind_speed : xarray.DataArray
+        Wind speed at the reference height.
+    h : Quantified
+        Height at which to compute the wind speed.
+    h_r : Quantified
+        Reference height.
+    method : {"power_law"}
+        Method to use. Currently only "power_law" is implemented.
+    kwds : dict
+        Additional keyword arguments to pass to the method. For `power_law`, this is `alpha`, which takes a default
```
Comment (laura-vanvliet) [inline xclim/indices/_conversion.py:2047]: Could add some additional clarification, if desired: "For `power_law`, this is `alpha`, which takes a default value of 1/7, but is highly variable based on topography, surface cover and atmospheric stability. "

File: tests/test_indices.py [hunk 3]
```diff
@@ -3626,3 +3626,24 @@ def test_late_frost_days_lat(self, tasmin_series):
...
+        np.testing.assert_allclose(v, a * 10 ** (1 / 7))
+
+
+class TestWindPowerPotential:
+    def test_simple(self, sfcWind_series):
+        a = [2, 6, 20, 30]
+        p = xci.wind_power_potential(
+            sfcWind_series(a, units="m/s"), cut_in="4 m/s", rated="8 m/s"
+        )
+        np.testing.assert_allclose(p, [0, 0.5**3, 1, 0])
```
Comment (laura-vanvliet) [inline tests/test_indices.py:3644]: I assume 0.5**3 is the same as [ (v - ui) / (ur - ui) ]**3 for these specific values, but flagging just in case :)

**Review by laura-vanvliet — CHANGES_REQUESTED** [review]
Hi @huard, I think overall this looks great! There is one important change due to another typo in the Tobin (2018) paper, and some suggestions for the function descriptors.

File: xclim/indices/_conversion.py [hunk 4]
```diff
@@ -2017,3 +2019,157 @@ def mean_radiant_temperature(
...
+    return out
+
+
+@vectorize
+def _wind_power_factor(v, cut_in, rated, cut_out):
+    """Wind power factor function"""
+    if v < cut_in:
+        return 0.0
+    if v < rated:
+        return ((v - cut_in) / (rated - cut_in)) ** 3
```
[the remark this answers is not in the document]
Comment (huard) [reply xclim/indices/_conversion.py:2172]: Got it, done.

File: xclim/indices/_conversion.py [hunk 5]
```diff
@@ -2017,3 +2019,157 @@ def mean_radiant_temperature(
...
+    by the cut-in wind speed (:math:`u_i`), the rated speed (:math:`u_r`) and the cut-out speed (:math:`u_o`).
+    Power production is zero for wind speeds below the cut-in speed, increases cubically between the cut-in
+    and rated speed, is constant between the rated and cut-out speed, and is zero for wind speeds above the cut-out
+    speed to avoid damage to the turbine :cite:p:`tobin_2018`:
+
+    .. math::
+
+        \begin{cases}
+        0,  &  v < u_i \\
+        (v - u_i)^3 / (u_r - u_i)^3,  & u_i ≤ v < u_r \\
```
[the remark this answers is not in the document]
Comment (huard) [reply xclim/indices/_conversion.py:2131]: Done

File: xclim/indices/_conversion.py (hunk 1 above)
Comment (huard) [reply xclim/indices/_conversion.py:2139]: Changed to 

Note that the temporal resolution of the wind speed time series has a significant influence on the results,
    even when aggregated at longer time scales. Total annual power production will differ substantially if estimated
    from hourly or daily wind speeds, due to the non-linearity of the production factor.

Is this what you meant ?

File: tests/test_indices.py (hunk 3 above)
Comment (huard) [reply tests/test_indices.py:3644]: Yeah it did, but not now with the change in the formula.

File: xclim/indices/_conversion.py [hunk 6]
```diff
@@ -5,7 +5,7 @@
 
 import numpy as np
 import xarray as xr
-from numba import float32, float64, vectorize  # noqa
+from numba import float32, float64, njit, vectorize  # noqa
```
Comment (aulemahal) [inline xclim/indices/_conversion.py:8]:
```suggestion
from numba import float32, float64, vectorize  # noqa
```
I think this change is now superfluous ?

**Review by aulemahal — APPROVED** [review]
Looks good!

May be you could add real tests in `test_atmos` ?

**Comment by laura-vanvliet** [comment]
Hi David, sorry for the long delay. I think the original wording was better. I was referring to, when summing up to create an annual power production estimate, they need to multiple by 3 if they use 3-hourly data, etc. But I'm not sure if that note is worth including.

File: xclim/indices/_conversion.py (hunk 6 above)
Comment (huard) [reply xclim/indices/_conversion.py:8]: Indeed !

**Comment by huard** [comment]
Ok, changed the wording of the docstring to simplify, and added a note explaining how to aggregate instantaneous power potential at lower frequencies. Tests have also been improved in the process, including one demoing the conversion to daily power production in MWh from 3-hourly values. 

Also fixed a bug in to_agg_units, but I note there are no explicit tests for this function.

**Pull request merged**